walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,73 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Iterator;
+import java.util.TreeMap;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    Collection<String> attendees = request.getAttendees();
+    TreeMap<TimeRange, Event> conflicts = 
+        new TreeMap<>(TimeRange.ORDER_BY_START);
+
+    //go through the events collection to find events with attendees from the
+    //request ie potential conflicts
+    Iterator eventItr = events.iterator();
+    while(eventItr.hasNext()) {
+      Event event = (Event)eventItr.next();
+      
+      //go through each event's attendees
+      Iterator attendeeItr = event.getAttendees().iterator();
+      while(attendeeItr.hasNext()){
+        String attendee = (String)attendeeItr.next();
+
+        //check if each attendee is in the attendees list for the new request
+        if (attendees.contains(attendee)) {
+          conflicts.put(event.getWhen(), event);
+          break;
+        }
+      }
+    }
+
+    //Iterate through conflicts by start time
```

Comment:
This is nitpicking, but be sure to be consistent in your comments with whether or not you're capitalizing and using punctuation. Also, not strictly necessary, but I find a space between "//" and the start of the comment makes it a little more readable!